File: py/nAryMaxDepth.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution:
    def maxDepth(self, root: 'Node') -> int:
        #for i in range(len(root.children)):
        if root == None: return 0
        if not root.children: return 1
        return 1+ max([self.maxDepth(child) for child in root.children])

File: js/nAryMaxDepth.js
/* // Definition for a Node.
* function Node(val,children) {
*    this.val = val;
*    this.children = children;
* };
*/

// this is almost as gold as Python's version with list comprehension
// basically you accummulate over the recursive results, but only maintaining the MAX value

var maxDepth = function(root) {
   if (!root) return 0;
   const max = root.children.reduce((max, child) => {
       return Math.max(max, maxDepth(child))
   }, 0);
   return max + 1;
};